File: sentimental-credit/credit.py
def main():
    # Main func

    while True:
        try:
            snum = input("Number: ")
            num = int(snum)
        except ValueError:
            continue
        else:
            break
    print(brand(check(num), num))


def check(n):
    # sum
    sum = 0

    # Each other digit double sum digits
    for i in range(2, len(str(n)) + 1, 2):
        try:
            sum += int(str(((n % (10 ** i)) // 10 ** (i-1)) * 2)[1])
            sum += int(str(((n % (10 ** i)) // 10 ** (i-1)) * 2)[0])
        except IndexError:
            sum += int(str(((n % (10 ** i)) // 10 ** (i-1)) * 2)[0])

    # sum of total
    if (len(str(n)) % 2) == 1:
        for i in range(0, len(str(n)), 2):
            sum += int(str(n)[i])
    else:
        for i in range(1, len(str(n)), 2):
            sum += int(str(n)[i])

    # Sum modulo 10
    check = sum % 10
    value = not check

    # return
    return value


def brand(cs, num):
    # func to print desired values
    snum = str(num)
    leng = len(snum)

    # cinditions
    if cs:
        if (leng == 15) and (snum[0] == '3') and ((snum[1] == '4') or (snum[1] == '7')):
            return "AMEX"
        elif (leng == 16) and (snum[0] == '5') and (int(snum[1]) >= 1) and (int(snum[1]) <= 5):
            return "MASTERCARD"
        elif ((leng == 16) or (leng == 13)) and (snum[0] == '4'):
            return "VISA"
    return "INVALID"


if __name__ == "__main__":
    main()

File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int get_digits(long a); // prototype for getting digits
int num_digits(long c); // prototype for getting num of digits
long power(long aa, int bb); // prototype for power function

int main(void)
{
    long x;
    do
    {
        x = get_long("Enter credit card number: ");
    }
    while (!(x >= 0));
    int alg = get_digits(x);
    int num = num_digits(x);
    int pre = x / power(10, num - 2);
    if (alg == 0)
    {
        if (num == 15 && (pre == 34 || pre == 37))
        {
            printf("AMEX\n");
        }
        else if (num == 16 && pre >= 51 && pre <= 55)
        {
            printf("MASTERCARD\n");
        }
        else if ((num == 13 || num == 16) && pre / 10 == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
    //printf("%i", get_digits(x));
}

int get_digits(long a)
{
    int i = 0; // digit itself
    long j = 1; // power of 10
    int b = 1; // even digits detector
    int sum_odd = 0; // sum of odd digits
    int sum_even = 0; // sum of even digits
    int total = 0; // total of sums
    while (a / j > 0)
    {
        i = (a / j) % 10;
        if (b % 2 == 0)
        {
            i *= 2;
            sum_even += ((i % 10) + ((i / 10)  % 10));
        }
        else
        {
            sum_odd += i;
        }
        //printf("%i\n", i);
        j *= 10;
        b++;
    }
    total = (sum_even + sum_odd) % 10;
    return total;
}



int num_digits(long c)
{
    int i = 0; // digit counter
    long j = 1; // power of 10
    while (c / j > 0)
    {
        i++;
        j *= 10;
    }
    return i;
}



long power(long aa, int bb)
{
    long ans = aa;
    for (int i = 1; i < bb; i++)
    {
        ans *= aa;
    }
    return ans;

}